### PR TITLE
[DCMAW-8782] Configure async backend api to send metrics to Dynatrace

### PR DESCRIPTION
**Pull request by mthkbarber**
​DCMAW-8782

### What changed
Update backend-api template.yaml to allow lambda functions to send metrics to Dynatrace.
Update test for new global environment variables

### Why did it change
We are creating Dynatrace dashboards for the async resources.

Stack successfully deployed:

image

Tests successfully run:
image

image

Dynatrace gathering metrics:
image

### CODE REVIEW

File: backend-api/template.yaml [hunk 1]
```diff
@@ -32,6 +32,18 @@ Parameters:
...
 Mappings:
+  EnvironmentConfiguration:
+    dev:
+      dynatraceSecretArn: arn:aws:secretsmanager:eu-west-2:216552277552:secret:DynatraceNonProductionVariables
+    build:
+      dynatraceSecretArn: arn:aws:secretsmanager:eu-west-2:216552277552:secret:DynatraceNonProductionVariables
+    staging:
+      dynatraceSecretArn: arn:aws:secretsmanager:eu-west-2:216552277552:secret:DynatraceNonProductionVariables
+    integration:
+      dynatraceSecretArn: arn:aws:secretsmanager:eu-west-2:216552277552:secret:DynatraceNonProductionVariables
```
Comment (sandymay-dd) [inline backend-api/template.yaml:43]: Just _double checking_ this shouldn't be ProductionVariables?

File: backend-api/template.yaml [hunk 2]
```diff
@@ -210,6 +222,27 @@ Globals:
...
+        DT_LOG_COLLECTION_AUTH_TOKEN: !Sub
+          - '{{resolve:secretsmanager:${SecretArn}:SecretString:DT_LOG_COLLECTION_AUTH_TOKEN}}'
+          - SecretArn: !FindInMap [ EnvironmentConfiguration, !Ref Environment, dynatraceSecretArn ]
+        DT_TENANT: !Sub
+          - '{{resolve:secretsmanager:${SecretArn}:SecretString:DT_TENANT}}'
+          - SecretArn: !FindInMap [ EnvironmentConfiguration, !Ref Environment, dynatraceSecretArn ]
+        DT_OPEN_TELEMETRY_ENABLE_INTEGRATION: "true"
+    Layers: 
+      - !Sub
+        - '{{resolve:secretsmanager:${SecretArn}:SecretString:NODEJS_LAYER}}' # or NODEJS_LAYER or PYTHON_LAYER
```
Comment (sandymay-dd) [inline backend-api/template.yaml:244]: nit: the comment can probably go here

File: backend-api/template.yaml (hunk 1 above)
Comment (mthkbarber) [reply backend-api/template.yaml:43]: I don't think so, we don't tend to use Int values in the Production Dynatrace environment and this is the same configuration as [mobile-id-check-backend](https://github.com/govuk-one-login/mobile-id-check-backend/blob/main/di-ipv-dca-oauth/template.yaml)

File: backend-api/template.yaml (hunk 2 above)
Comment (mthkbarber) [reply backend-api/template.yaml]: Done.

**Review by kikidawson-gds — APPROVED** [review]

**Pull request merged**